// einkauf-lib/rezepte.js
/**
 * 
 * Diese Datei beinhaltet die Rezepte, die die Anwendung kennt. 
 * 
 * 
 */

var publicApi;


var Rezepte = 
    [
        {
            rezeptNummer: "1",
            name: "Gurkenwurstbrot",
            zutaten:
            [
                { anzahl: 2, einheit: "Scheiben", artikel: "Brot" },
                { anzahl: 2, einheit: "Scheiben", artikel: "Wurst" },
                { anzahl: 7, einheit: "Scheiben", artikel: "Gurke" }
            ]
        },
        {
            rezeptNummer: "2",
            name: "Nudeln-Bolognese",
            zutaten:
            [
                { anzahl: 250, einheit: "Gramm", artikel: "Nudeln" },
                { anzahl: 1, einheit: "Glas", artikel: "Bolognesesoße" },
            ]
        },
        {
            rezeptNummer: "3",
            name: "Nudeln-Basilico",
            zutaten:
            [
                { anzahl: 250, einheit: "Gramm", artikel: "Nudeln" },
                { anzahl: 1, einheit: "Glas", artikel: "Basilico-Soße" },
            ]
        },
        {
            rezeptNummer: "4",
            name: "Würstchen mit Ketchup",
            zutaten:
            [
                { anzahl: 2, einheit: "Scheiben", artikel: "Brot" },
                { anzahl: 2, einheit: "Stück", artikel: "Würstchen" },
                { anzahl: 20, einheit: "ml", artikel: "Ketchup" },                
            ]
        },
        {
            rezeptNummer: "5",
            name: "Pfannkuchen",
            zutaten:
            [
                { anzahl: 200, einheit: "g", artikel: "Mehl" },
                { anzahl: 5, einheit: "Stück", artikel: "Eier" },
                { anzahl: 400, einheit: "ml", artikel: "Milch" },   
                { anzahl: 25, einheit: "g", artikel: "Zucker" },                             
            ]
        },
        {
            rezeptNummer: "6",
            name: "Wurstkartoffeln",
            zutaten:
            [
                { anzahl: 500, einheit: "g", artikel: "Kartoffeln" },
                { anzahl: 5, einheit: "Stück", artikel: "Würstchen" },
                { anzahl: 1, einheit: "Stück", artikel: "Zwiebel" },   
                { anzahl: 1, einheit: "Stück", artikel: "Knoblauchzehe" },    
                { anzahl: 1, einheit: "tl", artikel: "Paprikapulver"},
                { anzahl: 1, einheit: "tl", artikel: "Majoran"}
            ]
        },
        {
            rezeptNummer: "7",
            name: "Champignon-Porree-Käse-Curry-Suppe",
            zutaten:
            [
                { anzahl: 400, einheit: "g", artikel: "Porree" },
                { anzahl: 400, einheit: "g", artikel: "Champignons" },
                { anzahl: 400, einheit: "ml", artikel: "Milch" },   
                { anzahl: 0.4, einheit: "Stück", artikel: "Große Zwiebel" },                             
                { anzahl: 200, einheit: "g", artikel: "Hackfleisch" },
                { anzahl: 320, einheit: "g", artikel: "Schmelzkäse" },
                { anzahl: 0.6, einheit: "l", artikel: "Gemüsebrühe" },
                { anzahl: 0.8, einheit: "el, gest.", artikel: "Curry" }
            ]
        },   
        {
            rezeptNummer: "8",
            name: "Bananen-Nuss-Brot",
            zutaten:
            [
                { anzahl: 1, einheit: "Scheiben", artikel: "Brot" },
                { anzahl: 13, einheit: "g", artikel: "Dunkles Mandelmuß" },
                { anzahl: 1, einheit: "g", artikel: "Agavendicksaft" },
                { anzahl: 12, einheit: "Scheiben", artikel: "Banane" }
            ]
        },  
        {
            rezeptNummer: "9",
            name: "Brot alla Max (zum selber Backen)",
            zutaten:
            [
                { anzahl: 1000, einheit: "g", artikel: "Vollkorn-Dinkel-Mehl" },
                { anzahl: 34, einheit: "g", artikel: "Backpulver" },
                { anzahl: 15, einheit: "g", artikel: "Salz" },
                { anzahl: 18, einheit: "g", artikel: "Zucker" }
            ]
        },          
    ];

function IstRezeptBekannt(nameOderRezeptnummer) {
    for ( var i = 0; i < Rezepte.length; i++ ) {
        if ( Rezepte[i].name === nameOderRezeptnummer || 
             Rezepte[i].rezeptNummer === nameOderRezeptnummer ) {
                 return true;
             }
    }

    return false;
}

function RezeptNachNameOderRezeptnummer(nameOderRezeptnummer) {
    for ( var i = 0; i < Rezepte.length; i++ ) {
        if ( Rezepte[i].name === nameOderRezeptnummer || 
             Rezepte[i].rezeptNummer === nameOderRezeptnummer ) {
                 return Rezepte[i];
             }
    }  

    return undefined;  
}


publicApi = {
    Rezepte : Rezepte,
    IstRezeptBekannt : IstRezeptBekannt,
    Rezept : RezeptNachNameOderRezeptnummer
};

module.exports = publicApi;
